### DISCUSSION TITLE
Trigger's which are not in the repo are also disabled and not enabled after deployment

### DISCUSSION THREAD

**Discussion by rohan-citiustech** [Azure-Player/azure.datafactory.tools#239] [Show and tell]
#### I have 4 triggers each trigger reside in a different repo. let's name the trigger as A,B,C&D. 

- Assume i am deploying a repo which contains only trigger A

- **Before deployment state**: All 4 triggers A,B,C,D are in started state. My repo also has the runtimestate : "started"
- **After deploying state**: Trigger A is in started state but all other trigger B,C,D goes to in the disabled state  
- similarly if i deploy the repo with trigger B. Then after deployment trigger A,C&D are in disabled state only B will be in enable state
- **I feel the tool only restarts the trigger's which are available in the repo while deploying and hence it ignores the existing trigger which are not in the repo** but then i didn't understand why it disables the other 3 triggers or the triggers which are not in the repo
- I have to manually start the disabled triggers

**To Reproduce**

Create 2 repo's each having one trigger. Deploy 1st repo, the trigger from the 1st repo gets deployed in started state. Now deploy the 2nd repo after deployment you could see that 2nd repo's trigger is in started state but now the 1st trigger is in disabled state.

**Expected behaviour**
Triggers not in the repo should not be touched. And only triggers which are in the repo should be restarted  

**Version of PowerShell module of:**
 - v0.99

**Comment by NowinskiK** [comment]
That's intentional behaviour as the tools does not suppose to work with various repos as a source.
If this is your case - it means you're doing something specific, you have edge case, and hence additional efforts from your side is required. 
There are two comprehensive tables explain #adftools behaviour in terms of trigger:
https://github.com/SQLPlayer/azure.datafactory.tools#selective-deployment-triggers-and-logic

You have 2 options:
1) Change your approach and start using one repo only OR
2) Set `StartStopTriggers = $false` and build your custom logic for this
